includes/reader-activation/sync/class-woocommerce.php

```diff
@@ -285,6 +285,12 @@ private static function get_order_metadata( $order, $payment_page_url = false )
 					$metadata['product_name'] = reset( $subscription_order_items )->get_name();
 				}
 			}
+
+			// Record the cancellation reason if the subscription was cancelled.
+			$cancellation_reason = $current_subscription->get_meta( WooCommerce_Subscriptions::CANCELLATION_REASON_META_KEY, '' );
```

Comment:
I'm getting an error that I believe is coming from this line

```
Class "Newspack\Reader_Activation\Sync\WooCommerce_Subscriptions" not found
```

The meta was not added neither to the subscription nor to the ESP

Reply:
Ah. Good catch! Fixed in https://github.com/Automattic/newspack-plugin/pull/3568/commits/2a51cf403dab1e46609e7eb615499e62d3d02f86

> The meta was not added neither to the subscription nor to the ESP

Strange that the meta was not being added to the subscription though. I just updated the testing instructions, but make sure you've defined the ff constant to see the changes at work.

Also, I've run into this problem in the past so thought I'd mention it here as well, but if you have a contact with too many meta fields in MC, the newer fields will simply not be added. You can still check the logs for what will be synced with Newspack logging enabled in these cases:

![Screenshot 2024-11-26 at 16 52 16](https://github.com/user-attachments/assets/85712883-217d-4dee-bb33-f4c4bf1b0878)

Reply:
> make sure you've defined the ff constant to see the changes at work.

I'd never make such a mistake........ :grimacing:

Reply:
> Also, I've run into this problem in the past so thought I'd mention it here as well, but if you have a contact with too many meta fields in MC, the newer fields will simply not be added. 

Yes. This is a known issue. It's a limitation we have to deal with Mailchimp, that's why we can choose which fields we want. If you scroll the logs up a little you will see that there was an attempt to create the merge field and it failed because it reached the limit.